File: src/main/scala/vm/vagrant/util/VagrantException.scala
package vm.vagrant.util


/**
  * Default Exception for the vagrant-binding. Any Exception in Ruby / Vagrant is wrapped in a VagrantException
  *
  * @author oliver.ziegert
  *
  */
@SerialVersionUID(1L)
class VagrantException(message: String, cause: Throwable) extends RuntimeException (message, cause) {
  def this(message: String) {
    this(message, null)
  }

  def this(cause: Throwable) {
    this(null, cause)
  }
}

File: src/main/java/vm/vagrant/util/VagrantException.java
package vm.vagrant.util;

import java.io.Serializable;

/**
 * Default Exception for the vagrant-binding. Any Exception in Ruby / Vagrant is wrapped in a VagrantException
 * @author hendrikebbers
 *
 */
public class VagrantException extends RuntimeException implements Serializable {

	private static final long serialVersionUID = 1L;

	public VagrantException() {
        super();
    }

    public VagrantException(String message) {
        super(message);
    }

    public VagrantException(String message, Throwable cause) {
        super(message, cause);
    }

    public VagrantException(Throwable cause) {
        super(cause);
    }
	
}
